WalletWasabi.Fluent/ViewModels/AddWallet/ConnectHardwareWalletViewModel.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Input;
+using DynamicData;
+using NBitcoin;
+using ReactiveUI;
+using WalletWasabi.Gui;
+using WalletWasabi.Gui.Tabs.WalletManager.HardwareWallets;
+using WalletWasabi.Hwi;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.Fluent.ViewModels.AddWallet
+{
+	public class ConnectHardwareWalletViewModel : RoutableViewModel
+	{
+		private readonly Network _network;
+		private readonly List<Wallet> _currentWallets;
+		private CancellationTokenSource _searchHardwareWalletCts;
+		private HardwareWalletViewModel _selectedHardwareWallet;
+
+		public ConnectHardwareWalletViewModel(NavigationStateViewModel navigationState, Network network,
+			WalletManager walletManager) : base(navigationState, NavigationTarget.DialogScreen)
+		{
+			_network = network;
+			_currentWallets = walletManager.GetWallets().ToList();
+			_searchHardwareWalletCts = new CancellationTokenSource(TimeSpan.FromMinutes(3));
+			HardwareWallets = new ObservableCollection<HardwareWalletViewModel>();
+
+			EnumerateIfHardwareWalletsAsync();
+
+			CancelCommand = ReactiveCommand.Create(() => navigationState.DialogScreen?.Invoke().Router.NavigationStack.Clear());
```

Comment:
`navigationState.DialogScreen?.Invoke().Router.NavigationStack.Clear()`
to
`ClearNavigation()`

Reply:
Done